File: apps/android/VLinkerOBD/app/src/main/java/com/jslee1972/vlinkerobd/ui/ParameterGroups.kt
package com.jslee1972.vlinkerobd.ui

/**
 * Section labels and display order for the "即時參數" grid. Which field belongs to which group is
 * now data (each PID's own `group` in shared/vehicle-profiles/ JSON files, schema v5 — see
 * [ParameterMetadata]); this object only keeps the fixed set of section names and the order they
 * render in, which is a layout concern, not vehicle data.
 */
object ParameterGroups {
    const val TRIP_COMPUTER = "行車電腦"
    const val ENGINE = "引擎與動力"
    const val TEMPERATURE = "溫度"
    const val PRESSURE = "壓力"
    const val FUEL_EMISSIONS = "燃油與排放"
    const val ELECTRICAL = "電力與診斷"
    const val BRAND_SPECIFIC = "廠牌專屬"

    /** Render order for the sections — trip computer and engine first as the most driving-relevant. */
    val displayOrder = listOf(TRIP_COMPUTER, ENGINE, TEMPERATURE, PRESSURE, FUEL_EMISSIONS, ELECTRICAL, BRAND_SPECIFIC)

    /** The derived (non-PID) fields DashboardViewModel.updateTripComputer produces — also needed
     * by the custom-section field picker's "known fields" list, so exposed rather than duplicated. */
    val TRIP_COMPUTER_FIELDS = listOf(
        "instantFuelConsumption", "averageFuelConsumption", "acceleration", "tripDistance", "tripDuration",
    )

    /** Restricted to universal (never brand-specific) fields so it's meaningful on any vehicle.
     * Only ever consulted by [SharedPreferencesCustomSectionStore] when nothing has been
     * configured yet — the user can freely add/remove from here afterward, including clearing it
     * entirely via the picker's "全部移除". */
    val DEFAULT_CUSTOM_FIELDS = listOf(
        "coolantTempC", "fuelLevelPercent", "controlModuleVoltage", "engineLoadPercent",
        "instantFuelConsumption", "averageFuelConsumption", "tripDistance", "ambientAirTempC",
    )

    /** The driving-dynamics ring gauge's own center legend defaults to 水溫 + 油量 — the two
     * figures that used to be hardcoded there before it became user-configurable. Only ever
     * consulted before the user has picked their own pair (see [RingLegendFieldsStore]). */
    val DEFAULT_RING_LEGEND_FIELDS = listOf("coolantTempC", "fuelLevelPercent")
}

File: apps/ios/VLinkerOBD/ParameterGroups.swift
import Foundation

/// Section labels and display order for the live-readings grid. Which field belongs to which
/// group is data (each PID's own `group` in shared/vehicle-profiles JSON, schema v5 — see
/// ``ParameterMetadata``); this only holds the fixed set of section names and render order, a
/// layout concern shared with the Android port (`ParameterGroups.kt`).
enum ParameterGroups {
    static let tripComputer = "行車電腦"
    static let engine = "引擎與動力"
    static let temperature = "溫度"
    static let pressure = "壓力"
    static let fuelEmissions = "燃油與排放"
    static let electrical = "電力與診斷"
    static let brandSpecific = "廠牌專屬"

    /// Render order for the sections — trip computer and engine first as the most driving-relevant.
    static let displayOrder = [tripComputer, engine, temperature, pressure, fuelEmissions, electrical, brandSpecific]

    /// The derived (non-PID) fields the trip computer produces — also needed by the custom-section
    /// field picker's "known fields" list.
    static let tripComputerFields = ["instantFuelConsumption", "averageFuelConsumption", "acceleration", "tripDistance", "tripDuration"]

    /// Seeded into the custom section / driving-dynamics side panels the first time the app runs
    /// (before the user has picked anything) — a reasonable "most owners care about this" set,
    /// restricted to universal (never brand-specific) fields so it's meaningful on any vehicle.
    /// The user can freely add/remove from here afterward, including clearing it entirely; this is
    /// only ever consulted when nothing has been configured yet (see `CustomSectionStore`).
    static let defaultCustomFields: Set<String> = [
        "coolantTempC", "fuelLevelPercent", "controlModuleVoltage", "engineLoadPercent",
        "instantFuelConsumption", "averageFuelConsumption", "tripDistance", "ambientAirTempC",
    ]
}
